lib/Queue.rb

```diff
@@ -0,0 +1,38 @@
+
+class Queue
```

Comment:
I really like how you made separate Stack and Queue files. It creates DRY-er code, and also guides how you treat the tree. Nice!